### PR TITLE
Make sure to set discusion's categoryID when Vanilla.Categories.Use is false

### PR DESCRIPTION
**Pull request by DaazKu**
Fixes https://github.com/vanilla/vanilla/issues/5203

### CODE REVIEW

File: applications/vanilla/controllers/class.postcontroller.php [hunk 1]
```diff
@@ -172,17 +172,22 @@ public function discussion($CategoryUrlCode = '') {
...
+            // See if we should fill the CategoryID value.
             $AllowedCategories = CategoryModel::getByPermission(
                 'Discussions.Add',
                 $this->Form->getValue('CategoryID', $this->CategoryID),
                 ['Archived' => 0, 'AllowDiscussions' => 1],
                 ['AllowedDiscussionTypes' => $this->Data['Type']]
             );
-            if (count($AllowedCategories) == 1) {
-                $AllowedCategory = array_pop($AllowedCategories);
+            $nbsAllowedCategories = count($AllowedCategories);
```
Comment (linc) [inline applications/vanilla/controllers/class.postcontroller.php]: nbs? When did we start doing weird abbreviations in var names?

File: applications/vanilla/controllers/class.postcontroller.php [hunk 2]
```diff
@@ -172,17 +172,22 @@ public function discussion($CategoryUrlCode = '') {
 
         touchValue('Type', $this->Data, 'Discussion');
 
-        // See if we should hide the category dropdown.
-        if ($this->ShowCategorySelector) {
+        if (!$UseCategories || $this->ShowCategorySelector) {
```
Comment (linc) [inline applications/vanilla/controllers/class.postcontroller.php]: Didn't we just start removing this logic condition everywhere in prep for deprecating that config?

Comment (DaazKu) [inline applications/vanilla/controllers/class.postcontroller.php]: The config is still used. We need to either no use it anywhere or respect it until we remove it for real.

File: applications/vanilla/controllers/class.postcontroller.php (hunk 1 above)
Comment (DaazKu) [inline applications/vanilla/controllers/class.postcontroller.php]: How do you want me to call it? `$AllowedCategoriesCount`?
`$AllowedCategories` would be dumb here

Comment (linc) [inline applications/vanilla/controllers/class.postcontroller.php]: `$AllowedCategoriesCount` is fine. Or allowedCount. Or categoryCount. Or the inverse of any of those.

**Pull request merged**